contracts/exchange/contracts/src/MixinExchangeRichErrors.sol

```diff
@@ -20,13 +20,113 @@ pragma solidity ^0.5.9;
 
 import "@0x/contracts-utils/contracts/src/RichErrors.sol";
 import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
-import "./mixins/MExchangeRichErrors.sol";
 
 
 contract MixinExchangeRichErrors is
-    RichErrors,
-    MExchangeRichErrors
+    RichErrors
 {
+    enum AssetProxyDispatchErrorCodes {
+        INVALID_ASSET_DATA_LENGTH,
+        UNKNOWN_ASSET_PROXY
+    }
+
+    enum FillErrorCodes {
+        INVALID_TAKER_AMOUNT,
+        TAKER_OVERPAY,
+        OVERFILL,
+        INVALID_FILL_PRICE
+    }
+
+    enum SignatureErrorCodes {
+        BAD_SIGNATURE,
+        INVALID_LENGTH,
+        UNSUPPORTED,
+        ILLEGAL,
+        INAPPROPRIATE_SIGNATURE_TYPE
+    }
+
+    enum TransactionErrorCodes {
+        NO_REENTRANCY,
+        ALREADY_EXECUTED,
+        EXPIRED
+    }
+
+    // bytes4(keccak256("SignatureError(uint8,bytes32,address,bytes)"))
```

Comment:
It could be better to put these in a standalone contract. Especially since they're repeated below in `LibExchangeRichErrorDecoder.sol`.